### ISSUE TITLE
imdb and sst datasets do not use path argument for .splits() and .iters()

### ISSUE THREAD

**Issue by dhpollack** [pytorch/text#118]
The "path" argument is never considered when using the .splits() or .iters() functions

    imdb = datasets.IMDB("data")
    train, test = imdb.splits(TEXT, LABEL)

Logically, the above code should put the data in the "data" folder, but splits and iters use their own root folder (the "root" argument), so the data ends up in the current folder (".").

**Comment by bmccann** [comment]
What is the benefit of instantiating first? Does `train, test = IMDB.splits(TEXT, LABEL, root='data')` not do what you want?

**Comment by dhpollack** [comment]
There may not be any benefit, but it confused me because of how the torchvision datasets work.  The torchvision datasets tend to instantiate first and thus I instinctively tried that first and ran into errors.  Secondly, why have the "path" argument at all if you don't use it.

It's not wrong this way, but I found the inconsistency with torchvision confusing.  Plus it doesn't seem like it would be difficult to allow both options.

    def splits(..., root=None, ...)
        if self.path:
            root = self.path
        elif root is None:
            root = "."

Or just remove the "path" argument.

**Comment by jekbradbury** [comment]
`<dataset_class>.splits` always creates n Dataset objects, so it would be a little strange to create one first then throw it away and return n from an instance method. The API is fundamentally a little different from torchvision -- there's no concept here of instantiating a dataset reader, then using it n times.
